producer/db_adapter.go

```diff
@@ -44,14 +46,32 @@ func getMongoClient(url string, dbname string) error {
 	return errConnect
 }
 
-func ConnectMongo(url string, dbname string) {
+// Set AuthDBClient
+func setAuthDBClient(authurl string, authkeysdbname string) error {
+	var mClient, errConnect = mongoapi.NewMongoClient(authurl, authkeysdbname)
+	if mClient.Client != nil {
+		AuthDBClient = mClient
+		AuthDBClient.(*mongoapi.MongoClient).Client.Database(authkeysdbname)
+	}
+	return errConnect
+}
+
+func ConnectMongo(url string, dbname string, authurl string, authkeysdbname string) {
 	// Connect to MongoDB
+	if authurl == "" {
```

Comment:
I see you are setting default here if not available. ..better place would be to set default in config parsing code.

Reply:
Done